src/context/WalletProvider/config.ts

```diff
@@ -83,7 +86,6 @@ export interface SupportedWalletInfo {
   connectedWalletMenuInitialPath?: WalletConnectedRoutes
   connectedMenuComponent?: React.ComponentType<any>
 }
-
 export const SUPPORTED_WALLETS: Record<KeyManager, SupportedWalletInfo> = {
```

Comment:
you can then do the following instead of `Record<KeyManager, SupportedWalletInfo>`:

```typescript
type SupportedWalletInfo<T extends IWalletBaseAdapter> = {
  adapters: {
    loadAdapter: () => Promise<T>
  }[]
  supportsMobile?: 'browser' | 'app' | 'both'
  icon: ComponentWithAs<'svg', IconProps>
  name: string
  description?: string
  routes: RouteProps[]
  connectedWalletMenuRoutes?: RouteProps[]
  connectedWalletMenuInitialPath?: WalletConnectedRoutes
  connectedMenuComponent?: React.ComponentType<any>
}

type SupportedWalletInfoByKeyManager = {
  [KeyManager.Mobile]: SupportedWalletInfo<IMobileAdapter>
  [KeyManager.Native]: SupportedWalletInfo<INativeAdapter>
  [KeyManager.KeepKey]: SupportedWalletInfo<IKeepKeyAdapter>
  // ...
}
```

which then means when you use it you'll get the adapter type when loading it.

Similar in WalletProvider.tsx: 
`export type Adapters = Map<KeyManager, GenericAdapter[]>`

We can make it explicit so the whole app can take advantage of the types, rather than using a Map, make it an object and type it like so:
```typescript
type AdaptersByKeyManager = {
  [KeyManager.Mobile]: IMobileAdapter[]
  [KeyManager.Native]: INativeAdapter[]
  [KeyManager.KeepKey]: IKeepKeyAdapter[]
  // ...
}
```
```

Reply:
Will leave it for a part 4 PR alongside hdwallet bumpy interface exports